src/templates/speciality.js

```diff
@@ -65,22 +708,31 @@ const Speciality = ({
   const books = getExternalType(speciality, `Book`)
   const specialityEvents = events ? getSpecialityEvents(title, events) : []
 
+  // required: Head, IntroSection, TrainingSection, GetInTouch
+  // optional: ProjectsSection, Community, Talks, BlogListing, Tutorials, Books
+
+  // if (validateSpecialityCore(flattenedSpeciality)) {
+
   return (
     <Layout backgroundColor="blue" location={location}>
-      <Head page={speciality} />
-      <IntroSection speciality={speciality} />
-      <ProjectsSection
-        related={relatedProjects}
-        title={title}
-        clients={clients}
-      />
-      <TrainingSection speciality={speciality} />
-      <CommunitySection
-        background={communityBackground}
-        logo={communityLogo}
-        text={communityText}
-        title={title}
-      />
+      <Head page={flattenedSpeciality} />
+      <IntroSection speciality={flattenedSpeciality} />
+      {!!(
+        (relatedProjects &&
+          relatedProjects.length &&
+          validateProjects(relatedProjects)) ||
+        (clients && clients.length && validateClients(clients))
```

Comment:
Where we validate the schema like this, do we not call `.validate(obj)` on the method returned from `ajv.comile()`? instead of just calling the return function...

Reply:
Looks like you can also just do `ajv.validate(schema, data);`

Reply:
Also, looks like you don't need `clients && clients.length` as ajv will handle all of that for us

https://runkit.com/omonk/5d1acb94846c9f00132410b7

Reply:
So... you'd this it'd would work, but if I try your suggested approach (to just rely on ajv validation), I get this if clients and relatedProjects are empty arrays (which I guess is unlikely to happen but it's one way of simulating no data)

![image](https://user-images.githubusercontent.com/31079643/60502496-dc447b80-9cb5-11e9-86ab-d0a312f04c59.png)

Reply:
Should I assume that there will either be no data at all (ie no empty array), or there will be data? Given those assumptions we can just use the schema validation as conditional.

Reply:
ahh I see what you mean, yep that makes sense.

if there are no related projects it will be `null`, remember GraphQL will return null for anything like `[]`

[query](http://localhost:8000/___graphql?query=%7B%0A%20%20contentfulSpeciality(slug%3A%20%7Beq%3A%20%22rxjs%22%7D)%20%7B%0A%20%20%20%20slug%0A%20%20%20%20relatedProjects%20%7B%0A%20%20%20%20%20%20...%20on%20ContentfulTemplatedCaseStudy%20%7B%0A%20%20%20%20%20%20%20%20__typename%0A%20%20%20%20%20%20%20%20title%0A%20%20%20%20%20%20%20%20slug%0A%20%20%20%20%20%20%20%20introSentence%0A%20%20%20%20%20%20%20%20posterColor%0A%20%20%20%20%20%20%20%20posterImage%20%7B%0A%20%20%20%20%20%20%20%20%20%20title%0A%20%20%20%20%20%20%20%20%20%20file%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20url%0A%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20...%20on%20ContentfulNonTemplatedCaseStudy%20%7B%0A%20%20%20%20%20%20%20%20__typename%0A%20%20%20%20%20%20%20%20title%0A%20%20%20%20%20%20%20%20slug%0A%20%20%20%20%20%20%20%20posterColor%0A%20%20%20%20%20%20%20%20posterImage%20%7B%0A%20%20%20%20%20%20%20%20%20%20title%0A%20%20%20%20%20%20%20%20%20%20file%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20url%0A%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D%0A&operationName=undefined)